sites/docs/pages/components/date-input/index.md

```diff
@@ -0,0 +1,302 @@
+            <DateInput
+                name=range_filtering_a_query
+                data={orders_by_day}
+                dates=day
+                range
+            />
+
+        <LineChart
+            data={filtered_query_ranged}
+            x=day
+            y=sales
+        />
+    </div>
+
+````markdown
+
+```sql filtered_query_ranged
+select 
+    *
+from ${orders_by_day}
+where day between '${inputs.range_filtering_a_query.start}' - '${inputs.range_filtering_a_query.end}'
+```
+
+<DateInput
+    name=range_filtering_a_query
+    data={orders_by_day}
+    dates=day
+    range
+/>
+
+<LineChart
+    data={filtered_query_ranged}
+    x=day
+    y=sales
+/>
+````
+</DocTab>
+
+### Default Value for Preset Ranges
+
+<DocTab>
+    <div slot='preview'>
+            <DateInput defaultValue={'Last 7 Days'} range/>
+    </div>
+
+````svelte
+<DateInput
+    name=name_of_date_range
+    defaultValue={'Last 7 Days'}
+    range
+/>
+````
+</DocTab>
+
+### Customizing Single Preset Ranges
+
+<DocTab>
+    <div slot='preview'>
+            <DateInput presetRanges={'Last 7 Days'} range/>
+    </div>
+
+```svelte
+<DateInput
+    name=name_of_date_range
+    range
+    presetRanges={'Last 7 Days'}
+/>
+```
+</DocTab>
+
+### Customizing Multiple Preset Ranges
+
+<DocTab>
+    <div slot='preview'>
+            <DateInput presetRanges={['Last 7 Days', 'Last 3 Months', 'Year to Date', 'All Time']} range/>
+    </div>
+
+````svelte
+<DateInput
+    name=name_of_date_range
+    range
+    presetRanges={['Last 7 Days', 'Last 3 Months', 'Year to Date', 'All Time']}
+/>
+````
+</DocTab>
+
+### Manually Specifying a Range
+
+<DocTab>
+    <div slot='preview'>
+            <DateInput
+                name=manual_date_range
+                start=2019-01-01
+                end=2019-12-31
+                range
+            />
+    </div>
+
+```markdown
+<DateInput
+    name=manual_date_range
+    start=2019-01-01
+    end=2019-12-31
+    range
+/>
+```
+</DocTab>
+
+## Options
+
+<PropListing 
+    name="name"
+    description="Name of the DateInput, used to reference the selected values elsewhere as {`{inputs.name.start`} or {`inputs.name.end`}"
+    required=true
+    options="string"
+/>
+<PropListing 
+    name="data"
+    description="Query name, wrapped in curly braces"
+    options="query name"
+/>
+<PropListing 
+    name="range"
+    description="toggles between a ranged and single input calendar"
+    options={["true", "false"]}
+    default=false
+/>
+<PropListing 
+    name="dates"
+    description="Column name from the query containing Date Input to span"
+    options="column name"
+/>
+<PropListing 
+    name="start"
+    description="A manually specified start date to use for the range"
+    options="string formatted YYYY-MM-DD"
+/>
+<PropListing 
+    name="end"
+    description="A manually specified end date to use for the range"
+    options="string formatted YYYY-MM-DD"
+/>
+<PropListing 
+    name="title"
+    options="string"
```

Comment:
I think it looks odd to just have a floating number in a button. not obvious what to do without a title?

Maybe we can make this title a default?

I know @mcrascal has an different view so happy to discuss

Reply:
yeah it's definitely missing something, and easily overlooked on its own